src/StackExchange.Redis/ConnectionMultiplexer.cs

```diff
@@ -1731,7 +1731,18 @@ internal async Task<bool> ReconfigureAsync(bool first, bool reconfigureAll, LogP
                                 server.AutoConfigure(null);
                             }
                             log?.WriteLine($"Server endpoint {server.EndPoint} is in {server.ConnectionState}");
-                            available[i] = server.SendTracer(log);
+                            if (server.IsConnected)
+                            {
+                                // If we're connected, send the tracer immediately
+                                available[i] = server.SendTracer(log);
+                            }
+                            else
+                            {
+                                // Potential race: connection happens after we land in the else here
+                                var tcs = new TaskCompletionSource<bool>();
+                                server.FullyEstablished += (o, e) => tcs.TrySetResult(true);
```

Comment:
I'm a bit unclear on the lifetimes here, so I'll just ask the question to the experts - do we also need to unsubscribe the event handler? Or is it basically unimportant because server.FullyEstablished will only get invoked once anyway, and there's minimal garbage that can't be collected.

Reply:
Be sure to grab latest - very different now. However, something's not right, many connections are taking the full 5 seconds and it's not working as intended. Trying to refactor a bit further locally but...something is wrong here and not behaving as expected, causing the await to take the full 5 before succeeding.

Reply:
Hm. I don't see the bug yet. Was reverting the proposed fix for 1719 what broke it by any chance?

Reply:
Or could it be some chicken and egg problem where waiting for the state change causes it to never happen??

Reply:
@TimLovellSmith pushed another approach that's far cleaner I think - let's see what tests do here. Curious what you think of the structure - thoughts when you have time please?